web-module/src/main/java/by/tut/mdcatalog/project2/web/controller/ArticleController.java

```diff
@@ -2,39 +2,95 @@
 
 import by.tut.mdcatalog.project2.service.ArticleService;
 import by.tut.mdcatalog.project2.service.CommentService;
+import by.tut.mdcatalog.project2.service.UserService;
 import by.tut.mdcatalog.project2.service.model.ArticleDTO;
 import by.tut.mdcatalog.project2.service.model.CommentDTO;
+import by.tut.mdcatalog.project2.service.model.UserDTO;
+import org.springframework.security.core.Authentication;
+import org.springframework.security.core.context.SecurityContextHolder;
 import org.springframework.stereotype.Controller;
 import org.springframework.ui.Model;
 import org.springframework.ui.ModelMap;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.validation.BindingResult;
 import org.springframework.web.bind.annotation.GetMapping;
 import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestParam;
 
+import javax.validation.Valid;
+import java.util.Date;
 import java.util.List;
 
 @Controller
 public class ArticleController {
-
+    private static final Logger logger = LoggerFactory.getLogger(ItemAPIController.class);
     private final ArticleService articleService;
     private final CommentService commentService;
-    public ArticleController(ArticleService articleService, CommentService commentService) {
+    private final UserService userService;
+
+
+    public ArticleController(ArticleService articleService,
+                             CommentService commentService,
+                             UserService userService) {
         this.articleService = articleService;
         this.commentService = commentService;
+        this.userService = userService;
     }
 
     @GetMapping("/articles")
     public String getArticles(Model model) {
-        List<ArticleDTO> articlesDTO = articleService.getArticles();
+        List<ArticleDTO> articlesDTO = articleService.getAll();
         model.addAttribute("articles", articlesDTO);
         return "articles";
     }
 
     @GetMapping("/articles/{id}")
-    public String getArticle (@PathVariable(value = "id") Long id, ModelMap modelMap) {
+    public String getArticle(@PathVariable(value = "id") Long id, ModelMap modelMap) {
+        Authentication authentication = SecurityContextHolder.getContext().getAuthentication();
+        String username = authentication.getName();
+        UserDTO currentUserDTO = userService.getByUsername(username);
+
         ArticleDTO articleDTO = articleService.getById(id);
         List<CommentDTO> commentsDTO = commentService.getByArticleId(id);
         modelMap.addAttribute("article", articleDTO);
         modelMap.addAttribute("comments", commentsDTO);
+        modelMap.addAttribute("currentuserid", currentUserDTO.getId());
```

Comment:
We dont need to use currentUserId because we can get it constantly at any time

Reply:
done